### DISCUSSION TITLE
Not every Data is transferred to MongoDB

### DISCUSSION THREAD

**Discussion by BenRodProd** [neuefische/questions#591] [Web] [answered]
My first Question in the discussion-thread, hope I do it right...

When adding a Book, i want to write it in MongoDB, what does function surprisingly well, but in the same writing I want to change the existing "currentBook" to the bookTitle and it never changes... 

 ```javascript 
const updatedUserData = {
      ...userData[0],
      books: [...userData[0].books, newBook],
      currentBook: bookTitle,
    };
    const response = await fetch(`/api/users/${userData[0]._id}`, {
      method: "PUT",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(updatedUserData),
    });
    console.log(userData[0].currentBook);
    if (!response.ok) {
      console.error(`Error: ${response.status}`);
    } 
```

the content of "newBook" is written inside the "books"-Array, but the currentBook value does not change... Any Ideas why?
No error messages... just no change there...

Link to full repo:
https://github.com/BenRodProd/capstone-project/pull/47

Thank you for your time!!!

**Comment by onemanwenttomow** [comment, marked as answer]
So i just talked with Ben in person and the issue was that the userSchema was missing the `currentBook`

``` js 
import mongoose from "mongoose";

const { Schema } = mongoose;

const userSchema = new Schema({
  name: { type: String, required: true },
  books: { type: Array, required: true },
  sound: { type: Array, required: true },
  subtitle: { type: String, required: true },
});

const User = mongoose.models.User || mongoose.model("User", userSchema);

export default User;
```
